tools/targets/STM32_gen_PeripheralPins.py

```diff
@@ -873,8 +888,80 @@ def print_qspi(l):
     if ADD_DEVICE_IF:
         out_c_file.write( "#endif\n" )
 
+def print_usb(lst):
+    use_hs_in_fs = False
+    nb_loop = 1
+    inst = "USB_FS"
+    if lst == usb_otgfs_list:
```

Comment:
This is a list element comparison - use 'is' instead. However, the pattern of accessing a global variable from inside the function in this way makes the code harder to understand, harder to maintain and harder to test.